### ISSUE TITLE
Allow tolerations to be specified in cluster.yml for built-in addons

### ISSUE THREAD

**Issue by mkjpryor-stfc** [rancher/rke#1365]
The built-in addons, such as ingress, currently support `node_selector` for targeting specific nodes for deployment. It would be great if they also allowed the specification of `tolerations` to allow them to be scheduled on nodes with specific taints. 

I have two use cases:

1. For deployments on clouds where internet-facing IPs are limited, it would be useful to be able to deploy the ingress controller on the controlplane nodes so that ingress and the API can be exposed with one IP. This is not currently possible because while a node_selector can be added to target the controlplane nodes, the taint cannot be tolerated. 

2. The use of dedicated infrastructure nodes. This would also require custom taints for nodes (#157). In larger or production grade clusters, it is sometimes useful to have some workers dedicated as infrastructure nodes, e.g. for running the ingress controller, custom registries or a storage system like rook without interference from other workloads. This would require the ability to specify tolerations for the built-in addons in order to schedule them on these nodes.

**Comment by nicknezis** [comment]
Agreed! The two use cases are very similar to enterprise use cases we also have. I was trying to add support for specifying taints in the cluster.yml through these two pull requests (one for the types and another for rke): [rancher/types#829](https://github.com/rancher/types/pull/829) & #1349.

**Comment by Oats87** [comment]
Related: https://github.com/rancher/rke/issues/1066
